pex/pex_bootstrapper.py

```diff
@@ -576,12 +576,16 @@ def bootstrap_pex(
     # possible in the run.
     with ENV.patch(PEX_ROOT=pex_info.pex_root):
         if not execute:
-            for location in _activate_pex(entry_point, pex_info, venv_dir=venv_dir):
-                from pex.third_party import VendorImporter
+            locations = list(_activate_pex(entry_point, pex_info, venv_dir=venv_dir))
+            from pex.bootstrap import Bootstrap
+            from pex.third_party import VendorImporter, uninstall
 
+            for location in locations:
                 VendorImporter.install(
                     uninstallable=False, prefix="__pex__", path_items=["."], root=location
                 )
+            uninstall()
```

Comment:
It would be nice if `pex.pex.PEX` and this bit here shared a utility function, but I can also circle back and correct that if I'm motivated enough to put my money where my mouth is.

Either way, the activate, import, install, uninstall+demote is a finely ordered thing IIUC. In particular that Bootstrap and uninstall being imported when they are. If so, and you can shed light in a comment - great. If not, that's fine. This is hopefully pretty obviously fragile code.

Reply:
Unfortunately I cannot shed light as I have a very poor understanding of the underlying code. Feel free to edit the PR with a comment that you think is best.

Reply:
Yeah, I don't even know. I'll need to take time and experiment to come up with an explanation. I can do that after this merges.